sdks/java/extensions/sql/jdbc/src/test/java/org/apache/beam/sdk/extensions/sql/jdbc/BeamSqlLineIT.java

```diff
@@ -0,0 +1,310 @@
+                  BeamSqlLine.runSqlLine(args, null, outputStream, null);
+                  return toLines(outputStream);
+                });
+
+    // Wait 10 sec to allow creating a subscription.
+    Thread.sleep(10 * 1000);
+    List<PubsubMessage> messages =
+        ImmutableList.of(
+            message(
+                convertTimestampToMillis("2018-07-01 21:25:20"),
+                constructor.construct("id1", 1, 40.701, -74.001, 1000, 10, "enroute", 2)),
+            message(
+                convertTimestampToMillis("2018-07-01 21:26:06"),
+                constructor.construct("id2", 2, 40.702, -74.002, 1000, 10, "enroute", 4)),
+            message(
+                convertTimestampToMillis("2018-07-02 13:26:06"),
+                constructor.construct("id3", 3, 30, -72.32324, 2000, 20, "enroute", 7)),
+            message(
+                convertTimestampToMillis("2018-07-02 14:28:22"),
+                constructor.construct("id4", 4, 34, -73.32324, 2000, 20, "enroute", 8)));
+
+    eventsTopic.publish(messages);
+
+    assertThat(
+        Arrays.asList(
+            Arrays.asList("2018-07-01 21:25:20", "enroute", "40.701", "-74.001"),
+            Arrays.asList("2018-07-01 21:26:06", "enroute", "40.702", "-74.002")),
+        everyItem(IsIn.isOneOf(expectedResult.get(30, TimeUnit.SECONDS).toArray())));
+  }
+
+  @Test
+  public void testFixedWindow() throws Exception {
+    Future<List<List<String>>> expectedResult =
+        pool.submit(
+            (Callable)
+                () -> {
+                  String[] args =
+                      buildArgs(
+                          String.format(createPubsubTableStatement, publicTopic),
+                          setProject,
+                          fixedWindowQuery);
+                  ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                  BeamSqlLine.runSqlLine(args, null, outputStream, null);
+                  return toLines(outputStream);
+                });
+
+    // Wait 10 sec to allow creating a subscription.
+    Thread.sleep(10 * 1000);
+
+    // LIMIT 2 is supposed to be parsed into 6 lines.
+    assertTrue(expectedResult.get().size() == 6);
+  }
+
+  @Test
+  public void testSlidingWindow() throws Exception {
+    Future<List<List<String>>> expectedResult =
+        pool.submit(
+            (Callable)
+                () -> {
+                  String[] args =
+                      buildArgs(
+                          String.format(createPubsubTableStatement, publicTopic),
+                          setProject,
+                          slidingWindowQuery);
+                  ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                  BeamSqlLine.runSqlLine(args, null, outputStream, null);
+                  return toLines(outputStream);
+                });
+
+    // Wait 10 sec to allow creating a subscription.
+    Thread.sleep(10 * 1000);
```

Comment:
This one shouldn't need a thread or sleep at all.

Reply:
You are right. `testFixedWindow` does not need sleep neither as both tests are reading from the public topic.